addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -260,8 +256,55 @@ const Wysiwyg = Widget.extend({
         }
         const rpcMutex = new Mutex();
 
+        this._channelNames = new Set();
         this._peerToPeerLoading = new Promise(async (resolve) => {
-            let iceServers = await this._rpc({route: '/web_editor/get_ice_servers'});
+            const getChannelName = (signature) => `editor_collaboration_${signature}`;
+            const resetChannels = async () => {
+                // Retrie new infos
+                const infos = await this._rpc({
+                    route: '/web_editor/get_editor_collaboration_infos',
+                    params: {
+                        model_name: modelName,
+                        field_name: fieldName,
+                        res_id: resId,
+                    }
+                });
+                // The current signature is always the second entry in the array
+                // because the first and third entries are the previous and next
+                // broadcasting channels that we continue to listen in order to
+                // never miss a message whenever switching channels.
+                this._broadcastChannelSignature = infos.signatures[1];
+                // Remove all outdated channels. It is important for the
+                // security to stop listening the messages of outdated channels
+                // so that if a client had an access to a document revoked in
+                // the meantime, he will eventually be disconnected from others.
+                const channelNames = new Set(infos.signatures.map((signature) => getChannelName(signature)));
+                for (const channelName of this._channelNames) {
+                    if (!channelNames.has(channelName)) {
+                        this._channelNames.delete(channelName);
```

Comment:
`Set#delete` is actually not complete crap and will return whether the element was removed from the set (`true`).